docs: clearly deprecate Harness in the testing how-tos (#1508)
canonical/operator: docs/howto/write-scenario-tests-for-a-charm.md

(write-scenario-tests-for-a-charm)=
# How to write unit tests for a charm

> First of all, install the testing framework. To do this in a virtual environment
> while you are developing, you can do this with `pip` or any other package
> manager; for example:

Reviewer: Since this is the first time we're mentioning the testing framework on the page, shall we call it the "Ops testing framework"? (Also included some very minor nits, since I'm making a suggestion)

```suggestion
First of all, install the Ops testing framework. To do this in a virtual environment
while you are developing, use `pip` or another package
manager. For example:
```

Reviewer: Or "`ops` unit testing framework" for consistency with explanation/testing.md?

```
pip install ops[testing]
```

Normally, you'll include this in the dependency group for your unit tests, for
example in a test-requirements.txt file:

```text
ops[testing] ~= 2.17
```

Or in `pyproject.toml`:

```toml
[dependency-groups]
test = [
  "ops[testing] ~= 2.17",
]
```

Then, open a new `test_foo.py` file where you will put the test code.

```python
import ops
from ops import testing
```

Then declare a new charm type:
```python
class MyCharm(ops.CharmBase):
    pass        
```

And finally we can write a test function. The test code should use a `Context` object to encapsulate the charm type being tested (`MyCharm`) and any necessary metadata, then declare the initial `State` the charm will be presented when run, and `run` the context with an `event` and that initial state as parameters. 

In code:

```python
def test_charm_runs():
    # Arrange: 
    #  Create a Context to specify what code we will be running,
    ctx = testing.Context(MyCharm)
    #  and create a State to specify what simulated data the charm being run will access.

>     state_in = testing.State()

Reviewer 2: Might be good to have at least one parameter to `State()` for clarity, even just `leader=True`.
